yarn.lock

```diff
@@ -5628,6 +5645,28 @@ __metadata:
   languageName: node
   linkType: hard
 
+"@metamask/network-controller@npm:^18.1.2":
```

Comment:
this probably needs the patch for `@metamask/network-controller@18.1.0` ported?

Reply:
Hmmmm I can't figure out how this should be handled.

I looked at the [previous commit](https://github.com/MetaMask/metamask-extension/pull/23525/files) that touched this patch.

Which made me think I should do:
`yarn patch @metamask/network-controller@npm:18.1.2`

I did that and applied the existing patch logic.  But then it breaks the build with:
`Can't walk dependency graph: Cannot find module 'eth-block-tracker'`

Reply:
I think you're on the right track.

Since `@metamask/assets-controllers@30.0.0` depends on `@metamask/network-controller@18.1.2`, the update also implies updating to `@metamask/network-controller@18.1.2` and porting the patch.

The `eth-block-tracker` issue comes from the fact that the currently used version of `nonce-tracker` didn't correctly specify its peerdependency on `eth-block-tracker`, but was implicitly depending on it being available ([fixed here](https://github.com/MetaMask/nonce-tracker/pull/73)). Since `@metamask/network-controller` switches from `eth-block-tracker` to `@metamask/eth-block-tracker`, this happens.

A workaround might be to add `eth-block-tracker` as a dependency, though it would be preferred to properly migrate to `@metamask/eth-block-tracker` and `@metamask/nonce-tracker` (https://github.com/MetaMask/metamask-extension/pull/24861)

Reply:
I think after #24861 this might fall into place without need for any workarounds

Reply:
@legobeat i just rebased on the top of the branch #24861 , will do some tests now

Reply:
@salimtb :+1: 

See any blockers for #24861, considering those changes are now part of here?